ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/InsightsViewService.kt

```diff
@@ -93,11 +85,16 @@ class InsightsViewService(project: Project) : AbstractViewService(project) {
         }
     }
 
-    fun fetchForInsightStatus(methodInfo: MethodInfo, model: InsightsModel) {
+    fun updateModelInsightStatus(methodInfo: MethodInfo, model: InsightsModel) {
         val insightStatus = insightsProvider.getInsightStatus(methodInfo)
         val uiInsightStatus = toUiInsightStatus(insightStatus, methodInfo.hasRelatedCodeObjectIds())
+        // Unknown status returned on any backend exception. Unknown status displays 'loading...' msg on ui, so for now the default state when insight list empty and status unknown will be NoInsights
+        if (uiInsightStatus == UiInsightStatus.Unknown) {
+            model.status = UiInsightStatus.NoInsights
+        }else{
+            model.status = uiInsightStatus
+        }
```

Comment:
maybe add this logic to existing method `toUiInsightStatus` don't think we need for this extra check